// groundx-typescript-sdk/dist/client.js
"use strict";
/*
GroundX APIs

RAG Made Simple, Secure and Hallucination Free

The version of the OpenAPI document: 1.3.26
Contact: support@eyelevel.ai

NOTE: This file is auto generated by Konfig (https://konfigthis.com).
*/
Object.defineProperty(exports, "__esModule", { value: true });
exports.Groundx = void 0;
const api_1 = require("./api");
const configuration_1 = require("./configuration");
const client_custom_1 = require("./client-custom");
class Groundx extends client_custom_1.GroundxCustom {
    constructor(configurationParameters) {
        super(configurationParameters);
        const configuration = new configuration_1.Configuration(configurationParameters);
        this.buckets = new api_1.BucketsApi(configuration);
        this.customer = new api_1.CustomerApi(configuration);
        this.documents = new api_1.DocumentsApi(configuration);
        this.groups = new api_1.GroupsApi(configuration);
        this.health = new api_1.HealthApi(configuration);
        this.search = new api_1.SearchApi(configuration);
    }
}
exports.Groundx = Groundx;
